src/state/slices/limitOrderSlice/selectors.ts

```diff
@@ -0,0 +1,143 @@
+  },
+)
+
+export const selectActiveQuoteSellAsset = (state: ReduxState) => {
+  const sellAssetId = state.limitOrderSlice.activeQuote?.params.sellAssetId
+  return sellAssetId ? selectAssetById(state, sellAssetId) : undefined
+}
+
+export const selectActiveQuoteBuyAsset = (state: ReduxState) => {
+  const buyAssetId = state.limitOrderSlice.activeQuote?.params.buyAssetId
+  return buyAssetId ? selectAssetById(state, buyAssetId) : undefined
+}
+
+export const selectActiveQuoteFeeAsset = (state: ReduxState) => {
+  const sellAssetId = state.limitOrderSlice.activeQuote?.params.sellAssetId
+  return sellAssetId ? selectFeeAssetById(state, sellAssetId) : undefined
+}
+
+export const selectActiveQuoteSellAmountCryptoPrecision = createSelector(
+  selectActiveQuote,
+  selectActiveQuoteSellAsset,
+  (activeQuote, asset) => {
+    if (!activeQuote || !asset) return '0'
+    const { precision } = asset
+    return fromBaseUnit(activeQuote.response.quote.sellAmount, precision)
+  },
+)
+
+export const selectActiveQuoteBuyAmountCryptoPrecision = createSelector(
+  selectActiveQuote,
+  selectActiveQuoteBuyAsset,
+  (activeQuote, asset) => {
+    if (!activeQuote || !asset) return '0'
+    const { precision } = asset
+    // DANGER! DO NOT use the quote buy amount `activeQuote.response.quote.buyAmount`
+    // doing so will discard the user input limit and do a spot trade.
+    return fromBaseUnit(activeQuote.params.buyAmountCryptoBaseUnit, precision)
+  },
+)
+
+export const selectActiveQuoteNetworkFeeCryptoPrecision = createSelector(
+  selectActiveQuote,
+  selectActiveQuoteFeeAsset,
+  (activeQuote, asset) => {
+    if (!activeQuote || !asset) return '0'
+    const { precision } = asset
+    // CoW does not cost a network fee to submit, but the calcs are implemented as though they did
+    // in case we ever wire them in for a different protocol in the future.
+    const networkFee = '0'
+    return fromBaseUnit(networkFee, precision)
+  },
+)
+
+export const selectActiveQuoteSellAssetUserCurrencyRate = createSelector(
+  selectActiveQuoteSellAsset,
+  selectMarketDataUsd,
+  selectUserCurrencyToUsdRate,
+  (asset, marketDataUsd, userCurrencyToUsdRate) => {
+    const usdRate = marketDataUsd[asset?.assetId ?? '']?.price
+    return bnOrZero(usdRate).times(userCurrencyToUsdRate).toString()
+  },
+)
+
+export const selectActiveQuoteBuyAssetUserCurrencyRate = createSelector(
+  selectActiveQuoteBuyAsset,
+  selectMarketDataUsd,
+  selectUserCurrencyToUsdRate,
+  (asset, marketDataUsd, userCurrencyToUsdRate) => {
+    const usdRate = marketDataUsd[asset?.assetId ?? '']?.price
+    return bnOrZero(usdRate).times(userCurrencyToUsdRate).toString()
+  },
+)
+
+export const selectActiveQuoteSellAmountUserCurrency = createSelector(
+  selectActiveQuoteSellAmountCryptoPrecision,
+  selectActiveQuoteSellAssetUserCurrencyRate,
+  (amountCryptoPrecision, userCurrencyRate) => {
+    return bn(amountCryptoPrecision).times(userCurrencyRate).toString()
```

Comment:
q: don't we want to use toFixed instead of toString? Same for all the other occurences

Reply:
Yes youre right, I'll `Fix` :D

Reply:
a9906c7dcc920739cdb484a4f18f9b2296ff4857